modules/datadict/php/datadict.class.inc

```diff
@@ -113,16 +113,149 @@ class Datadict extends \NDB_Menu_Filter
     }
 
     /**
-    * Add query filters
-    *
-    * @param string $prepared_key the query string
-    * @param string $key          the key of filter
-    * @param string $val          the value of filter
-    *
-    * @note   Get base query
-    * @return string
-    * @access private
-    */
+     * Builds the filter array to be used in this filter menu. The end result
+     * should be that all the NDB_Menu_Filter attributes required for the query
+     * to work are set up.
+     *
+     * Generally, it does the following:
```

Comment:
i wish everybody commented their code this well :+1:

Reply:
I second that!